File: VB/Controllers/HomeController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Namespace DockingAddDynamicallyVBMvc.Controllers
	Public Class HomeController
		Inherits Controller
		Public Function Index() As ActionResult
			Return View()
        End Function
        Public Function DockPanelPartial(ByVal index As Integer) As ActionResult
            ViewData("index") = index
            Return PartialView()
        End Function
	End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System.Web.Mvc;

namespace DockingAddDynamicallyMvc.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }
        public ActionResult DockPanelPartial(int index) {
            ViewData["index"] = index;
            return PartialView();
        }
    }
}